# Task 1 - Data Collection

In [ ]:
# Loading libraries for manipulation and visualization of data

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [ ]:
# Read the file as pandas dataframe

df = pd.read_csv('/content/ecommerce_furniture_dataset_2024.csv')

In [ ]:
# Take a brief look at the data
df.head()

productTitle originalPrice    price  \
0  Dresser For Bedroom With 9 Fabric Drawers Ward...           NaN   $46.79   
1  Outdoor Conversation Set 4 Pieces Patio Furnit...           NaN  $169.72   
2  Desser For Bedroom With 7 Fabric Drawers Organ...         $78.4   $39.46   
3  Modern Accent Boucle Chair,Upholstered Tufted ...           NaN  $111.99   
4  Small Unit Simple Computer Desk Household Wood...        $48.82   $21.37   

   sold        tagText  
0   600  Free shipping  
1     0  Free shipping  
2     7  Free shipping  
3     0  Free shipping  
4     1  Free shipping

In [ ]:
# Get the dimentions of the dataframe
df.shape

(2000, 5)

In [ ]:
# Get the row names of the dataframe
df.index

RangeIndex(start=0, stop=2000, step=1)

In [ ]:
# Get the columns name of the dataframe
df.columns

Index(['productTitle', 'originalPrice', 'price', 'sold', 'tagText'], dtype='object')

In [ ]:
# Look at basic information about the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   productTitle   2000 non-null   object
 1   originalPrice  487 non-null    object
 2   price          2000 non-null   object
 3   sold           2000 non-null   int64 
 4   tagText        1997 non-null   object
dtypes: int64(1), object(4)
memory usage: 78.3+ KB


# Observations From Task 1

There are 2000 rows and 5 columns in the data, Each row contains the details of sales matrix.

The features in the dataset are:

*   Product and their original and discounted price
*   Sold Units and tagText

# Task 2 - Data Preprocessing

In [ ]:
# Finding null values if any
df.isnull().sum()

productTitle        0
originalPrice    1513
price               0
sold                0
tagText             3
dtype: int64

In [ ]:
# Dropping the null values
df.dropna(inplace=True)

In [ ]:
# New dimentions of the dataframe
df.shape

(487, 5)

In [ ]:
# df.drop(['originalPrice'],axis=1,inplace=True)

In [ ]:
df.head()

productTitle originalPrice    price  \
2   Desser For Bedroom With 7 Fabric Drawers Organ...         $78.4   $39.46   
4   Small Unit Simple Computer Desk Household Wood...        $48.82   $21.37   
5   3 Pieces Patio Furniture Set, Outdoor Swivel G...     $1,294.04   $123.4   
7   Living Room Furniture, Modern 3-Piece Includin...     $1,474.17  $734.58   
20  7-piece patio furniture set modular wicker out...       $671.47  $331.24   

    sold        tagText  
2      7  Free shipping  
4      1  Free shipping  
5     53  Free shipping  
7      5  Free shipping  
20    39  Free shipping

In [ ]:
df['tagText'].nunique()

3

In [ ]:
df['tagText'].value_counts()

tagText
Free shipping         485
+Shipping: $225.12      1
+Shipping: $2.91        1
Name: count, dtype: int64

# Observations From Task 2

Find out 1513 null values in originalPrice and 3 null values in tagText.


*   Removed all the null values
*   Data Cleaned and Ready for Analysis

